ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,20 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CXX=${CXX}
```

Comment:
You will need to handle `CC` in addition to `CXX` (we have a couple C dependencies which also deserve being ccache'd).